packages/grid/_modules_/grid/utils/tree/buildRowTree.ts

```diff
@@ -69,6 +70,25 @@ export const buildRowTree = (params: BuildRowTreeParams): GridRowGroupingResult
 
   const groupingCriteriaToIdTree: GridGroupingCriteriaToIdTree = {};
 
+  const isGroupExpandedByDefault = (node: GridRowTreeNodeConfig) => {
+    if (!node.children || !node.children.length) {
+      return false;
+    }
+
+    if (node.depth === treeDepth - 1) {
```

Comment:
I do not understand why you add this condition?

If `node.depth === treeDepth - 1` it is a leaf, so `node.children` should be undefined

Reply:
But leafs are prevented by the previous condition `!node.children || !node.children.length`

Reply:
You are right, fixed